MeesGame/Gameplay/GameObjects/TileField.cs

```diff
@@ -135,15 +135,41 @@ public void UpdateGraphics()
 
         public void RevealArea(Point a)
         {
+            revealAroundTile(a);
+
+            revealPath(a, new int[2] { -1, 0 });
+            revealPath(a, new int[2] { 1, 0 });
+            revealPath(a, new int[2] { 0, 1 });
+            revealPath(a, new int[2] { 0, -1 });
+        }
+
+        public void revealAroundTile(Point a)
+        {
             for (int i = a.X - 1; i <= a.X + 1; i++)
             {
                 for (int j = a.Y - 1; j <= a.Y + 1; j++)
                 {
-                    if (i >= 0 && i < this.Columns && j >= 0 && j < this.Rows) ((Tile)this.grid[i, j]).Revealed = true;
+                    if (i >= 0 && i < this.Columns && j >= 0 && j < this.Rows)
+                    {
+                        //using bitwise or-operator
+                        if (i <= a.X && j >= a.Y) { GetTile(i, j).Revealed = GetTile(i, j).Revealed | 1; }
+                        if (i <= a.X && j <= a.Y) { GetTile(i, j).Revealed = GetTile(i, j).Revealed | 2; }
+                        if (i >= a.X && j <= a.Y) { GetTile(i, j).Revealed = GetTile(i, j).Revealed | 4; }
+                        if (i >= a.X && j >= a.Y) { GetTile(i, j).Revealed = GetTile(i, j).Revealed | 8; }
+                    }
                 }
             }
         }
 
+        public void revealPath(Point a, int[] direction)
+        {
+            //check if you're still on the map, while that is the case and vision is not obstructed reveal the area around the tile.
+            while ((a.X += direction[0]) >= 0 && a.X < this.Columns && (a.Y += direction[1]) >= 0 && a.Y < this.Rows && !((Tile)this.grid[a.X, a.Y]).ObstructsVision)
```

Comment:
Put the different checks on multiple lines, this is too long to read.
Try checking if tiles are a WallTile instead of an extra property. If that works, code should be a bit cleaner.